konnect/login_service.go

```diff
@@ -54,17 +56,35 @@ func (s *AuthService) Login(ctx context.Context, email,
 	return authResponse, nil
 }
 
+func createAuthRequest(baseURL, email, password string) (*http.Request, error) {
+	var (
+		buf []byte
+		err error
+
+		body = map[string]string{
+			"username": email,
+			"password": password,
+		}
+	)
+
+	buf, err = json.Marshal(body)
+	if err != nil {
+		return nil, err
+	}
+
+	parts := strings.Split(baseURL, "api.konghq")
+	endpoint := baseEndpointUS + parts[len(parts)-1] + authEndpointV2
```

Comment:
I think it would be best to pull this out and unit test it thoroughly to just be sure. Breeding ground for a bug.